streams/src/test/java/org/apache/kafka/streams/processor/internals/RepartitionOptimizingTest.java

```diff
@@ -185,8 +185,8 @@ private void runTest(final String optimizationConfig, final int expectedNumberRe
             .filter((k, v) -> k.equals("A"), Named.as("join-filter"))
             .join(countStream, (v1, v2) -> v1 + ":" + v2.toString(),
                   JoinWindows.of(ofMillis(5000)),
-                  StreamJoined.<String, String, Long>with(Stores.inMemoryWindowStore("join-store", ofDays(1), ofMillis(10000), true),
-                                                          Stores.inMemoryWindowStore("other-join-store",  ofDays(1), ofMillis(10000), true))
+                  StreamJoined.<String, String, Long>with(Stores.inMemoryWindowStore("join-store", ofDays(1).plus(ofMillis(10000)), ofMillis(10000), true),
```

Comment:
This is the by-product of this change as we default grace to 24h, and retention to grace + window-size.

We can consider whether we keep the default grace at 24h cc @ableegoldman